### ISSUE TITLE
error: internal compiler error: unexpected panic

### ISSUE THREAD

**Issue by NotNorom** [rust-lang/rust-clippy#8430]
I ran into a clippy crash. First time, if I have done anything wrong reporting this issue, feel free to tell me! :D
The offending code can be found here: https://github.com/NotNorom/cutepokebot/blob/f25b709961c951ecadf759eb01136eac185f0dce/src/setup.rs#L64-L288

I am running the `stable-x86_64-unknown-linux-gnu` toolchain on arch linux.
Cargo version: `cargo 1.58.0 (f01b232bc 2022-01-19)`

Stacktrace:
```
thread 'rustc' panicked at 'called `Option::unwrap()` on a `None` value', /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/compiler/rustc_hir/src/definitions.rs:452:14
stack backtrace:
   0:     0x7f6b93648e8c - std::backtrace_rs::backtrace::libunwind::trace::h09f7e4e089375279
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x7f6b93648e8c - std::backtrace_rs::backtrace::trace_unsynchronized::h1ec96f1c7087094e
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f6b93648e8c - std::sys_common::backtrace::_print_fmt::h317b71fc9a5cf964
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x7f6b93648e8c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::he3555b48e7dfe7f0
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x7f6b936a655c - core::fmt::write::h513b07ca38f4fb1b
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/core/src/fmt/mod.rs:1149:17
   5:     0x7f6b936390d5 - std::io::Write::write_fmt::haf8c932b52111354
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/std/src/io/mod.rs:1697:15
   6:     0x7f6b9364bf30 - std::sys_common::backtrace::_print::h195c38364780a303
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x7f6b9364bf30 - std::sys_common::backtrace::print::hc09dfdea923b6730
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x7f6b9364bf30 - std::panicking::default_hook::{{closure}}::hb2e38ec0d91046a3
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/std/src/panicking.rs:211:50
   9:     0x7f6b9364bae5 - std::panicking::default_hook::h60284635b0ad54a8
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/std/src/panicking.rs:228:9
  10:     0x5634f2d1c3d9 - clippy_driver[3caad8ad6bf5a7b9]::ICE_HOOK::{closure#0}::{closure#0}
  11:     0x7f6b7b6ed813 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::hbb785100c76a11dc
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/alloc/src/boxed.rs:1708:9
  12:     0x7f6b7b61f51d - proc_macro::bridge::client::<impl proc_macro::bridge::Bridge>::enter::{{closure}}::{{closure}}::hf04db0b3829d84d0
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/proc_macro/src/bridge/client.rs:320:21
  13:     0x7f6b9364c749 - std::panicking::rust_panic_with_hook::ha677a669fb275654
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/std/src/panicking.rs:610:17
  14:     0x7f6b9364c1d2 - std::panicking::begin_panic_handler::{{closure}}::h976246fb95d93c31
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/std/src/panicking.rs:500:13
  15:     0x7f6b93649334 - std::sys_common::backtrace::__rust_end_short_backtrace::h38077ee5b7b9f99a
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/std/src/sys_common/backtrace.rs:139:18
  16:     0x7f6b9364c169 - rust_begin_unwind
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/std/src/panicking.rs:498:5
  17:     0x7f6b936104b1 - core::panicking::panic_fmt::h35f3a62252ba0fd2
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/core/src/panicking.rs:107:14
  18:     0x7f6b936103fd - core::panicking::panic::h86fc01e270142a61
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/core/src/panicking.rs:48:5
  19:     0x7f6b958dd93e - <rustc_middle[26370726012d1bf2]::ty::context::TyCtxt>::def_path_hash_to_def_id
  20:     0x7f6b95918182 - <rustc_query_system[3616b1a045b1904c]::dep_graph::dep_node::DepNode<rustc_middle[26370726012d1bf2]::dep_graph::dep_node::DepKind> as rustc_middle[26370726012d1bf2]::dep_graph::dep_node::DepNodeExt>::extract_def_id
  21:     0x7f6b95fc519e - rustc_query_impl[4fd9530611cf4d08]::query_callbacks::hir_owner::force_from_dep_node
  22:     0x7f6b962f1801 - <rustc_middle[26370726012d1bf2]::ty::context::TyCtxt as rustc_query_system[3616b1a045b1904c]::dep_graph::DepContext>::try_force_from_dep_node
  23:     0x7f6b955dee2c - <rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepGraph<rustc_middle[26370726012d1bf2]::dep_graph::dep_node::DepKind>>::try_mark_previous_green::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt>
  24:     0x7f6b955dedfd - <rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepGraph<rustc_middle[26370726012d1bf2]::dep_graph::dep_node::DepKind>>::try_mark_previous_green::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt>
  25:     0x7f6b955dedfd - <rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepGraph<rustc_middle[26370726012d1bf2]::dep_graph::dep_node::DepKind>>::try_mark_previous_green::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt>
  26:     0x7f6b95577674 - rustc_query_system[3616b1a045b1904c]::query::plumbing::try_load_from_disk_and_cache_in_memory::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, rustc_middle[26370726012d1bf2]::infer::canonical::Canonical<rustc_middle[26370726012d1bf2]::ty::ParamEnvAnd<rustc_middle[26370726012d1bf2]::ty::Predicate>>, core[7435606f1c80df2a]::result::Result<rustc_middle[26370726012d1bf2]::traits::select::EvaluationResult, rustc_middle[26370726012d1bf2]::traits::select::OverflowError>>
  27:     0x7f6b95545872 - rustc_query_system[3616b1a045b1904c]::query::plumbing::try_execute_query::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, rustc_query_system[3616b1a045b1904c]::query::caches::DefaultCache<rustc_middle[26370726012d1bf2]::infer::canonical::Canonical<rustc_middle[26370726012d1bf2]::ty::ParamEnvAnd<rustc_middle[26370726012d1bf2]::ty::Predicate>>, core[7435606f1c80df2a]::result::Result<rustc_middle[26370726012d1bf2]::traits::select::EvaluationResult, rustc_middle[26370726012d1bf2]::traits::select::OverflowError>>>
  28:     0x7f6b955baf37 - <rustc_query_impl[4fd9530611cf4d08]::Queries as rustc_middle[26370726012d1bf2]::ty::query::QueryEngine>::evaluate_obligation
  29:     0x7f6b957e2a2b - <rustc_infer[c3769667aa5ba053]::infer::InferCtxt as rustc_trait_selection[e183c0ec72eef9d2]::traits::query::evaluate_obligation::InferCtxtExt>::evaluate_obligation
  30:     0x7f6b957e2e30 - <rustc_infer[c3769667aa5ba053]::infer::InferCtxt as rustc_trait_selection[e183c0ec72eef9d2]::traits::query::evaluate_obligation::InferCtxtExt>::evaluate_obligation_no_overflow
  31:     0x7f6b957f773d - <rustc_trait_selection[e183c0ec72eef9d2]::traits::fulfill::FulfillProcessor>::process_trait_obligation
  32:     0x7f6b957f11a2 - <rustc_trait_selection[e183c0ec72eef9d2]::traits::fulfill::FulfillProcessor>::progress_changed_obligations
  33:     0x7f6b95820457 - <rustc_data_structures[a6a643d2c4658522]::obligation_forest::ObligationForest<rustc_trait_selection[e183c0ec72eef9d2]::traits::fulfill::PendingPredicateObligation>>::process_obligations::<rustc_trait_selection[e183c0ec72eef9d2]::traits::fulfill::FulfillProcessor, rustc_data_structures[a6a643d2c4658522]::obligation_forest::Outcome<rustc_trait_selection[e183c0ec72eef9d2]::traits::fulfill::PendingPredicateObligation, rustc_infer[c3769667aa5ba053]::traits::FulfillmentErrorCode>>
  34:     0x7f6b957f06d2 - <rustc_trait_selection[e183c0ec72eef9d2]::traits::fulfill::FulfillmentContext>::select
  35:     0x7f6b957f0f72 - <rustc_trait_selection[e183c0ec72eef9d2]::traits::fulfill::FulfillmentContext as rustc_infer[c3769667aa5ba053]::traits::engine::TraitEngine>::select_with_constness_where_possible
  36:     0x7f6b95168d75 - <rustc_typeck[17f4c348c1653d0a]::check::fn_ctxt::FnCtxt>::resolve_generator_interiors
  37:     0x7f6b951e17ac - <rustc_infer[c3769667aa5ba053]::infer::InferCtxtBuilder>::enter::<&rustc_middle[26370726012d1bf2]::ty::context::TypeckResults, <rustc_typeck[17f4c348c1653d0a]::check::inherited::InheritedBuilder>::enter<rustc_typeck[17f4c348c1653d0a]::check::typeck_with_fallback<rustc_typeck[17f4c348c1653d0a]::check::typeck::{closure#0}>::{closure#1}, &rustc_middle[26370726012d1bf2]::ty::context::TypeckResults>::{closure#0}>
  38:     0x7f6b951b8953 - rustc_typeck[17f4c348c1653d0a]::check::typeck
  39:     0x7f6b95fe04c3 - <rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepGraph<rustc_middle[26370726012d1bf2]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[26370726012d1bf2]::ty::context::TyCtxt, rustc_span[c29ad455a793da3]::def_id::LocalDefId, &rustc_middle[26370726012d1bf2]::ty::context::TypeckResults>
  40:     0x7f6b95f66a04 - rustc_data_structures[a6a643d2c4658522]::stack::ensure_sufficient_stack::<(&rustc_middle[26370726012d1bf2]::ty::context::TypeckResults, rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepNodeIndex), rustc_query_system[3616b1a045b1904c]::query::plumbing::execute_job<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, rustc_span[c29ad455a793da3]::def_id::LocalDefId, &rustc_middle[26370726012d1bf2]::ty::context::TypeckResults>::{closure#3}>
  41:     0x7f6b9555152c - rustc_query_system[3616b1a045b1904c]::query::plumbing::try_execute_query::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, rustc_query_system[3616b1a045b1904c]::query::caches::DefaultCache<rustc_span[c29ad455a793da3]::def_id::LocalDefId, &rustc_middle[26370726012d1bf2]::ty::context::TypeckResults>>
  42:     0x7f6b955b830c - <rustc_query_impl[4fd9530611cf4d08]::Queries as rustc_middle[26370726012d1bf2]::ty::query::QueryEngine>::typeck
  43:     0x7f6b958dba3c - <rustc_middle[26370726012d1bf2]::ty::context::TyCtxt>::typeck_opt_const_arg
  44:     0x7f6b953997e5 - rustc_mir_build[e9d23ada89acae2f]::build::mir_built
  45:     0x7f6b95fd4212 - <rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepGraph<rustc_middle[26370726012d1bf2]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[26370726012d1bf2]::ty::context::TyCtxt, rustc_middle[26370726012d1bf2]::ty::WithOptConstParam<rustc_span[c29ad455a793da3]::def_id::LocalDefId>, &rustc_data_structures[a6a643d2c4658522]::steal::Steal<rustc_middle[26370726012d1bf2]::mir::Body>>
  46:     0x7f6b95f659f9 - rustc_data_structures[a6a643d2c4658522]::stack::ensure_sufficient_stack::<(&rustc_data_structures[a6a643d2c4658522]::steal::Steal<rustc_middle[26370726012d1bf2]::mir::Body>, rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepNodeIndex), rustc_query_system[3616b1a045b1904c]::query::plumbing::execute_job<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, rustc_middle[26370726012d1bf2]::ty::WithOptConstParam<rustc_span[c29ad455a793da3]::def_id::LocalDefId>, &rustc_data_structures[a6a643d2c4658522]::steal::Steal<rustc_middle[26370726012d1bf2]::mir::Body>>::{closure#3}>
  47:     0x7f6b95542714 - rustc_query_system[3616b1a045b1904c]::query::plumbing::try_execute_query::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, rustc_query_system[3616b1a045b1904c]::query::caches::DefaultCache<rustc_middle[26370726012d1bf2]::ty::WithOptConstParam<rustc_span[c29ad455a793da3]::def_id::LocalDefId>, &rustc_data_structures[a6a643d2c4658522]::steal::Steal<rustc_middle[26370726012d1bf2]::mir::Body>>>
  48:     0x7f6b955b6c54 - <rustc_query_impl[4fd9530611cf4d08]::Queries as rustc_middle[26370726012d1bf2]::ty::query::QueryEngine>::mir_built
  49:     0x7f6b95135838 - rustc_mir_transform[4aee5ff76806a560]::check_unsafety::unsafety_check_result
  50:     0x7f6b9513196a - <rustc_mir_transform[4aee5ff76806a560]::check_unsafety::provide::{closure#0} as core[7435606f1c80df2a]::ops::function::FnOnce<(rustc_middle[26370726012d1bf2]::ty::context::TyCtxt, rustc_span[c29ad455a793da3]::def_id::LocalDefId)>>::call_once
  51:     0x7f6b95fe1543 - <rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepGraph<rustc_middle[26370726012d1bf2]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[26370726012d1bf2]::ty::context::TyCtxt, rustc_span[c29ad455a793da3]::def_id::LocalDefId, &rustc_middle[26370726012d1bf2]::mir::query::UnsafetyCheckResult>
  52:     0x7f6b95f66d24 - rustc_data_structures[a6a643d2c4658522]::stack::ensure_sufficient_stack::<(&rustc_middle[26370726012d1bf2]::mir::query::UnsafetyCheckResult, rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepNodeIndex), rustc_query_system[3616b1a045b1904c]::query::plumbing::execute_job<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, rustc_span[c29ad455a793da3]::def_id::LocalDefId, &rustc_middle[26370726012d1bf2]::mir::query::UnsafetyCheckResult>::{closure#3}>
  53:     0x7f6b955530bc - rustc_query_system[3616b1a045b1904c]::query::plumbing::try_execute_query::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, rustc_query_system[3616b1a045b1904c]::query::caches::DefaultCache<rustc_span[c29ad455a793da3]::def_id::LocalDefId, &rustc_middle[26370726012d1bf2]::mir::query::UnsafetyCheckResult>>
  54:     0x7f6b95e86116 - rustc_query_system[3616b1a045b1904c]::query::plumbing::force_query::<rustc_query_impl[4fd9530611cf4d08]::queries::unsafety_check_result, rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt>
  55:     0x7f6b95fc6e91 - rustc_query_impl[4fd9530611cf4d08]::query_callbacks::unsafety_check_result::force_from_dep_node
  56:     0x7f6b962f1801 - <rustc_middle[26370726012d1bf2]::ty::context::TyCtxt as rustc_query_system[3616b1a045b1904c]::dep_graph::DepContext>::try_force_from_dep_node
  57:     0x7f6b955dee2c - <rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepGraph<rustc_middle[26370726012d1bf2]::dep_graph::dep_node::DepKind>>::try_mark_previous_green::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt>
  58:     0x7f6b955dedfd - <rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepGraph<rustc_middle[26370726012d1bf2]::dep_graph::dep_node::DepKind>>::try_mark_previous_green::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt>
  59:     0x7f6b955dedfd - <rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepGraph<rustc_middle[26370726012d1bf2]::dep_graph::dep_node::DepKind>>::try_mark_previous_green::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt>
  60:     0x7f6b95effc4a - rustc_query_system[3616b1a045b1904c]::query::plumbing::try_load_from_disk_and_cache_in_memory::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, rustc_span[c29ad455a793da3]::def_id::LocalDefId, &rustc_middle[26370726012d1bf2]::mir::query::BorrowCheckResult>
  61:     0x7f6b955521d8 - rustc_query_system[3616b1a045b1904c]::query::plumbing::try_execute_query::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, rustc_query_system[3616b1a045b1904c]::query::caches::DefaultCache<rustc_span[c29ad455a793da3]::def_id::LocalDefId, &rustc_middle[26370726012d1bf2]::mir::query::BorrowCheckResult>>
  62:     0x7f6b9557dc73 - rustc_query_system[3616b1a045b1904c]::query::plumbing::get_query::<rustc_query_impl[4fd9530611cf4d08]::queries::mir_borrowck, rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt>
  63:     0x7f6b9522affd - rustc_typeck[17f4c348c1653d0a]::collect::type_of::type_of
  64:     0x7f6b95feed9a - <rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepGraph<rustc_middle[26370726012d1bf2]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[26370726012d1bf2]::ty::context::TyCtxt, rustc_span[c29ad455a793da3]::def_id::DefId, &rustc_middle[26370726012d1bf2]::ty::TyS>
  65:     0x7f6b95f662b7 - rustc_data_structures[a6a643d2c4658522]::stack::ensure_sufficient_stack::<(&rustc_middle[26370726012d1bf2]::ty::TyS, rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepNodeIndex), rustc_query_system[3616b1a045b1904c]::query::plumbing::execute_job<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, rustc_span[c29ad455a793da3]::def_id::DefId, &rustc_middle[26370726012d1bf2]::ty::TyS>::{closure#3}>
  66:     0x7f6b95563813 - rustc_query_system[3616b1a045b1904c]::query::plumbing::try_execute_query::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, rustc_query_system[3616b1a045b1904c]::query::caches::DefaultCache<rustc_span[c29ad455a793da3]::def_id::DefId, &rustc_middle[26370726012d1bf2]::ty::TyS>>
  67:     0x7f6b955b6536 - <rustc_query_impl[4fd9530611cf4d08]::Queries as rustc_middle[26370726012d1bf2]::ty::query::QueryEngine>::type_of
  68:     0x7f6b9523137e - rustc_typeck[17f4c348c1653d0a]::check::check::check_item_type
  69:     0x7f6b9520fbc5 - <rustc_middle[26370726012d1bf2]::hir::map::Map>::visit_item_likes_in_module::<rustc_typeck[17f4c348c1653d0a]::check::CheckItemTypesVisitor>
  70:     0x7f6b95cda18c - rustc_typeck[17f4c348c1653d0a]::check::check::check_mod_item_types
  71:     0x7f6b95fe2042 - <rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepGraph<rustc_middle[26370726012d1bf2]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[26370726012d1bf2]::ty::context::TyCtxt, rustc_span[c29ad455a793da3]::def_id::LocalDefId, ()>
  72:     0x7f6b95f68e84 - rustc_data_structures[a6a643d2c4658522]::stack::ensure_sufficient_stack::<((), rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepNodeIndex), rustc_query_system[3616b1a045b1904c]::query::plumbing::execute_job<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, rustc_span[c29ad455a793da3]::def_id::LocalDefId, ()>::{closure#3}>
  73:     0x7f6b95553e41 - rustc_query_system[3616b1a045b1904c]::query::plumbing::try_execute_query::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, rustc_query_system[3616b1a045b1904c]::query::caches::DefaultCache<rustc_span[c29ad455a793da3]::def_id::LocalDefId, ()>>
  74:     0x7f6b95f245f3 - rustc_query_system[3616b1a045b1904c]::query::plumbing::get_query::<rustc_query_impl[4fd9530611cf4d08]::queries::check_mod_item_types, rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt>
  75:     0x7f6b95cba7e8 - <rustc_middle[26370726012d1bf2]::hir::map::Map>::for_each_module::<rustc_typeck[17f4c348c1653d0a]::check_crate::{closure#6}::{closure#0}>
  76:     0x7f6b95c6360c - rustc_typeck[17f4c348c1653d0a]::check_crate
  77:     0x7f6b959cb8e0 - rustc_interface[2c1d684caca70d53]::passes::analysis
  78:     0x7f6b95ffdbb0 - <rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepGraph<rustc_middle[26370726012d1bf2]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[26370726012d1bf2]::ty::context::TyCtxt, (), core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>>
  79:     0x7f6b95f61bd3 - rustc_data_structures[a6a643d2c4658522]::stack::ensure_sufficient_stack::<(core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>, rustc_query_system[3616b1a045b1904c]::dep_graph::graph::DepNodeIndex), rustc_query_system[3616b1a045b1904c]::query::plumbing::execute_job<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, (), core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>>::{closure#3}>
  80:     0x7f6b95ecc79c - rustc_query_system[3616b1a045b1904c]::query::plumbing::try_execute_query::<rustc_query_impl[4fd9530611cf4d08]::plumbing::QueryCtxt, rustc_query_system[3616b1a045b1904c]::query::caches::DefaultCache<(), core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>>>
  81:     0x7f6b95f7a265 - <rustc_query_impl[4fd9530611cf4d08]::Queries as rustc_middle[26370726012d1bf2]::ty::query::QueryEngine>::analysis
  82:     0x7f6b959bf699 - <rustc_interface[2c1d684caca70d53]::passes::QueryContext>::enter::<rustc_driver[6ad7101a8fabd86e]::run_compiler::{closure#1}::{closure#2}::{closure#3}, core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>>
  83:     0x7f6b95996fd7 - <rustc_interface[2c1d684caca70d53]::interface::Compiler>::enter::<rustc_driver[6ad7101a8fabd86e]::run_compiler::{closure#1}::{closure#2}, core[7435606f1c80df2a]::result::Result<core[7435606f1c80df2a]::option::Option<rustc_interface[2c1d684caca70d53]::queries::Linker>, rustc_errors[5befc41d352b3f14]::ErrorReported>>
  84:     0x7f6b95992ed4 - rustc_span[c29ad455a793da3]::with_source_map::<core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>, rustc_interface[2c1d684caca70d53]::interface::create_compiler_and_run<core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>, rustc_driver[6ad7101a8fabd86e]::run_compiler::{closure#1}>::{closure#1}>
  85:     0x7f6b95997e2f - <scoped_tls[66f88778b59ce329]::ScopedKey<rustc_span[c29ad455a793da3]::SessionGlobals>>::set::<rustc_interface[2c1d684caca70d53]::util::setup_callbacks_and_run_in_thread_pool_with_globals<rustc_interface[2c1d684caca70d53]::interface::run_compiler<core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>, rustc_driver[6ad7101a8fabd86e]::run_compiler::{closure#1}>::{closure#0}, core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>>::{closure#0}::{closure#0}, core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>>
  86:     0x7f6b95995f65 - std[b2a2bae5f6ca25d7]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[2c1d684caca70d53]::util::setup_callbacks_and_run_in_thread_pool_with_globals<rustc_interface[2c1d684caca70d53]::interface::run_compiler<core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>, rustc_driver[6ad7101a8fabd86e]::run_compiler::{closure#1}>::{closure#0}, core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>>::{closure#0}, core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>>
  87:     0x7f6b959c015a - <<std[b2a2bae5f6ca25d7]::thread::Builder>::spawn_unchecked<rustc_interface[2c1d684caca70d53]::util::setup_callbacks_and_run_in_thread_pool_with_globals<rustc_interface[2c1d684caca70d53]::interface::run_compiler<core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>, rustc_driver[6ad7101a8fabd86e]::run_compiler::{closure#1}>::{closure#0}, core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>>::{closure#0}, core[7435606f1c80df2a]::result::Result<(), rustc_errors[5befc41d352b3f14]::ErrorReported>>::{closure#1} as core[7435606f1c80df2a]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  88:     0x7f6b93657fc3 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hcbc6d2d80772be64
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/alloc/src/boxed.rs:1694:9
  89:     0x7f6b93657fc3 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h9bffa2ca65a1d6e6
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/alloc/src/boxed.rs:1694:9
  90:     0x7f6b93657fc3 - std::sys::unix::thread::Thread::new::thread_start::ha678a8b0caec8f55
                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b/library/std/src/sys/unix/thread.rs:106:17
  91:     0x7f6b93549259 - start_thread
  92:     0x7f6b9332c5e3 - __GI___clone
  93:                0x0 - <unknown>

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust-clippy/issues/new

note: Clippy version: clippy 0.1.58 (db9d1b2 2022-01-20)

query stack during panic:
#0 [evaluate_obligation] evaluating trait selection obligation `impl core::future::future::Future<Output = ()>: core::marker::Send`
#1 [typeck] type-checking `setup::<impl at src/setup.rs:64:1: 288:2>::start`
#2 [mir_built] building MIR for `setup::<impl at src/setup.rs:64:1: 288:2>::start`
#3 [unsafety_check_result] unsafety-checking `setup::<impl at src/setup.rs:64:1: 288:2>::start`
#4 [mir_borrowck] borrow-checking `setup::<impl at src/setup.rs:64:1: 288:2>::start`
#5 [type_of] computing type of `setup::<impl at src/setup.rs:64:1: 288:2>::start::{opaque#0}`
#6 [check_mod_item_types] checking item types in module `setup`
#7 [analysis] running analysis passes on this crate
end of query stack
```

**Comment by dswij** [comment]
I couldn't reproduce this on a clean build from your repo. I suppose this might have been a problem with incremental builds.

**Comment by giraffate** [comment]
Thanks for the report!

It's a duplicate of https://github.com/rust-lang/rust-clippy/issues/8076, so I'm closing this.

**Comment by NotNorom** [comment]
Albright, thanks for the feedback! :)
